### PR TITLE
Improved the pr template

### PR DESCRIPTION
**Pull request by grzesiek2010**
@lognaturel 
During the meeting we had yesterday we were talking about providing more info for QAs so I think it's a good idea to add a new section in the pr template.

### CODE REVIEW

**Comment by lognaturel** [comment]
Thanks, @grzesiek2010! I'm hesitant to add another section to the list because the longer it becomes the less likely people are to fill it out thoughtfully. Plus, the issue should generally describe reproduction steps and reviewers should help make sure the info is there.

That said, I think a reminder is helpful. How about changing `Do we need any specific form for testing your changes? If so, please attach one` to `Please describe any special steps needed to verify your change and attach a form if needed.`

https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#the-review-process already has a good checklist for reviewers and I think we just haven't been careful enough about making sure the PR description is complete. I'll commit to being more cautious and certainly appreciate a poke if I miss something. @mmarciniak90 if you do see a PR for which it's not clear what verification is needed, write a comment right away so we can get a sense of where there isn't enough detail.

**Comment by lognaturel** [comment]
I think @ggalmazor is thinking along the same lines. Let's have the conversation here and then make the templates consistent across the repos.

**Comment by ggalmazor** [comment]
Hi, @grzesiek2010 & @lognaturel!

**TL;DR**: I'd add a new section before all other with the question: "Please, describe with detail any new, removed or changed visible user features this PR brings. Describe all other non-visible changes to the code as well."

I like the existing section that forces contributors to think about a way to test their work and explain it. Many times, one will want to just commit something without testing it and having to explain how they tested it has the probable side effect of that stuff getting tested by them.

In my case, I often forget about describing the changes my PRs bring and, honestly, I remember being surprised by not seeing any section on the template to describe them at the beginning (then I forgot about it until now). I think that the first thing a contributor should do is just that. I know a new first section for that would help me not forget about it :)

In my experience, people who implement something are the worst ones to test it because they're (we're) inevitably biased. It doesn't have to do with them being unprofessional, lazy or dumb. It's just that their mindset is often focused on making things work, not finding what's broken. Both things might seem the same, but they're not.

I think that asking for specific QA testing steps can be counterproductive because there's high risk of contaminating the manual testing with this bias.

Also, by adding a specific section about QA we are sending the unconscious message that "quality is a stage of software development", and I don't like that implication. Quality must be implemented at all stages (I'm the first one struggling with this). 

Ideally, contributors can write automated unit and outside-in tests, reviewers can discuss their work, ask for new tests and offer help, and manual testers can search for broken stuff by semi-randomly using the software. All these activities are what ensure quality along the chain.

**Comment by lognaturel** [comment]
Hello people of the past! Finally have a moment to give this some thought.

> "Please, describe with detail any new, removed or changed visible user features this PR brings. Describe all other non-visible changes to the code as well."

This kind of information should be covered by the risk section. Hopefully when we think about risk we think about anything that would affect users. I do see that folks are not very good at filing out that section ("None" is generally a funny answer unless it's just a change to a test or something like that) so perhaps we can replace the risk section with something like what @ggalmazor is suggesting?

**Comment by ggalmazor** [comment]
> replace the risk section

I'll give that my :+1: :)
